web-console/knikubevirt/snapshots/snapshots.md

```diff
@@ -0,0 +1,72 @@
+# Virtual Machines Snapshots
+
+Snapshots allow users to create a copy of the VM disks and memory, mainly for restore and backup purposes. The snapshots are taken live or offline depending on if the VM is running or not. 
+This document describes the snapshot design flow.  
+
+
+
+
+## From VM List 
+
+![Offline VM list - snapshot action from a single VM kebab](img/1-00.jpg)
+When clicking on the kebab menu in the VM list, the user will be able to take a snapshot of the VM, meaning he will copy the disks, memory content, and metadata for restoring, backup, creating templates and cloning VMs.
+
+![Offline Snapshot modal - default](img/1-01.jpg)
+In the basic modal, the snapshot will take all current configuration and selected parameters.
+The user must add a descriptive name.
+![Offline Snapshot modal - selecting disks](img/1-02.jpg)
+In the disks field, the user will be able to decide if to include all disks or just some of them and if to include memory as part of the snapshot or not.
+![Offline Snapshot modal - disks customization](img/1-03.jpg)
+![Offline Snapshot modal - disks customized](img/1-04.jpg)
+
+![Offline Snapshotting status indication](img/1-05.jpg)
+![Offline Snapshotting status indication on click and kebab mode](img/1-06.jpg)
+
+Any virtual machine that has a snapshot in progress will show this in the state:
+When in this progress mode all actions would be disabled and the “Stop snapshot” would be available.
+
+
+## VM page - snapshots tab
+
+![Offline Shanpshot tab](img/2-00.jpg)
```

Comment:
The columns for "CPU: 10GB" and "Size: 2 CPU" don't make sense in these screens. I assume they were swapped. Since CPU and NICs/Networks are always going to be the same for every snapshot (they can't be changed by the user), it seems like those columns can be removed too.

Will the "Memory" of the snapshot always be the same 16GB? Or does the snapshot only include something like 2.37GB if that's all the VM is using at the time the snapshot is taken? Might be worth checking with devs.

Reply:
@yuvalgalanti  what do you think?

Reply:
I discussed this with Yuval, apparently, this should mostly indicate if it does or doesn't include memory (we are not sure if it's unique or the same of each time).
Adding an alternative in the next commit